api/controllers/requests-api-controller.ts

```diff
@@ -0,0 +1,149 @@
+import db from '../db/conn';
+import { Request, Response } from 'express';
+import { Collection, WithId, ObjectId } from 'mongodb';
+import RequestStates from '../constants/RequestStates';
+import { z } from 'zod';
+
+// All functions use requests collection
+const collection : Collection<RequestRecord> = db.collection<RequestRecord>('requests');
+
+// Schema to check for employee id when coming in as url params
+const idSchema = z.string().length(24);
+
+// The record expected from the requests collection
+export interface RequestRecord {
+  _id: ObjectId,
+  lateEntry: boolean,
+  firstName: string,
+  lastName: string,
+  employeeId: number,
+  idir: string,
+  itemsPurchased: Array<string>,
+  totalCost: number,
+  purchaseDate: string,
+  attachReceipts: Array<object>,
+  approvalDate: string,
+  attachApproval: Array<object>,
+  submit: boolean,
+  submissionDate: string,
+  state: RequestStates
+}
+
+// Allows for a reduced version of each document to be returned from Mongo
+const minimalProjection = {
+  firstName: 1,
+  lastName: 1,
+  itemsPurchased: 1,
+  totalCost: 1,
+  purchaseDate: 1,
+  submissionDate: 1,
+  state: 1
+}
+
+// Get all request records
+export const getAllRequests = async (req: Request, res: Response) => {
+  const { minimal } = req.query;
+  try {
+    // TODO: Is there a more memory-efficient solution? Like only returning the first X records?
```

Comment:
Something like this? https://isotropic.co/how-to-implement-pagination-in-mongodb/